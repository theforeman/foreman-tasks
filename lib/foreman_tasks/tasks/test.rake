namespace :test do
  task :foreman_tasks => 'db:test:prepare' do
    test_task = Rake::TestTask.new('foreman_tasks_test_task') do |t|
      t.libs << ['test', "#{ForemanTasks::Engine.root}/test"]
      t.test_files = ["#{ForemanTasks::Engine.root}/test/**/*_test.rb"]
      t.verbose = true
      t.warning = false
    end

    Rake::Task[test_task.name].invoke
  end
end

namespace :foreman_tasks do
  task :rubocop do
    begin
      require 'rubocop/rake_task'
      RuboCop::RakeTask.new(:rubocop_foreman_tasks) do |task|
	task.patterns = ["#{ForemanTasks::Engine.root}/app/**/*.rb",
		  "#{ForemanTasks::Engine.root}/lib/**/*.rb",
		  "#{ForemanTasks::Engine.root}/test/**/*.rb"]
      end
    rescue
      puts 'Rubocop not loaded.'
    end

    Rake::Task['rubocop_foreman_tasks'].invoke
  end
end

Rake::Task[:test].enhance do
  Rake::Task['test:foreman_tasks'].invoke
end

load 'tasks/jenkins.rake'
if Rake::Task.task_defined?(:'jenkins:unit')
  Rake::Task['jenkins:unit'].enhance ['foreman_tasks:rubocop', 'test:foreman_tasks']
end
