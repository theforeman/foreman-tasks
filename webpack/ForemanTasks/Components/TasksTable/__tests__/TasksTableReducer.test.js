import Immutable from 'seamless-immutable';
import { testReducerSnapshotWithFixtures } from '@theforeman/test';
import {
  TASKS_TABLE_ID,
  TASKS_TABLE_SET_SORT,
  TASKS_TABLE_SET_PAGINATION,
  SELECT_ROWS,
  UNSELECT_ROWS,
  UNSELECT_ALL_ROWS,
  UPDATE_CLICKED,
  SELECT_ALL_ROWS,
  OPEN_SELECT_ALL,
} from '../TasksTableConstants';
import reducer from '../TasksTableReducer';

const fixtures = {
  'should return the initial state': {},
  'should handle TASKS_TABLE_SUCCESS': {
    action: {
      type: `${TASKS_TABLE_ID}_SUCCESS`,
      response: {
        subtotal: 120,
        page: 3,
        per_page: 12,
      },
    },
  },
  'should handle TASKS_TABLE_SET_SORT': {
    action: {
      type: TASKS_TABLE_SET_SORT,
      payload: { by: 'a', order: 'b' },
    },
  },

  'should handle TASKS_TABLE_SET_PAGINATION': {
    action: {
      type: TASKS_TABLE_SET_PAGINATION,
      payload: { page: 4, perPage: 7 },
    },
  },
  'should handle SELECT_ROWS': {
    action: {
      type: SELECT_ROWS,
      payload: [1, 2, 3, 4, 5, 6, 7],
    },
  },
  'should handle UNSELECT_ROWS': {
    action: {
      type: UNSELECT_ROWS,
      payload: [{ id: 4 }],
    },
  },
  'should handle UNSELECT_ALL_ROWS': {
    action: {
      type: UNSELECT_ALL_ROWS,
    },
  },
  'should handle UPDATE_CLICKED': {
    action: {
      type: UPDATE_CLICKED,
      payload: { clicked: 'task' },
    },
  },
  'should handle SELECT_ALL_ROWS': {
    action: {
      type: SELECT_ALL_ROWS,
      payload: { clicked: 'task' },
    },
  },
  'should handle OPEN_SELECT_ALL': {
    action: {
      type: OPEN_SELECT_ALL,
    },
  },
  'should handle UNSELECT_ROWS with all rows selected': {
    state: Immutable({
      tasksTableQuery: { allRowsSelected: true, selectedRows: [3, 4, 5] },
    }),
    action: {
      type: UNSELECT_ROWS,
      payload: { id: [4], results: [{ id: 3 }, { id: 4 }, { id: 5 }] },
    },
  },
};

describe('TasksTableReducer reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
