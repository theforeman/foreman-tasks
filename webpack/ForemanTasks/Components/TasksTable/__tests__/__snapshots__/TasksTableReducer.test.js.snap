// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksTableReducer reducer should handle OPEN_SELECT_ALL 1`] = `
Object {
  "tasksTableQuery": Object {
    "selectedRows": Array [],
    "showSelectAll": true,
  },
}
`;

exports[`TasksTableReducer reducer should handle SELECT_ALL_ROWS 1`] = `
Object {
  "tasksTableQuery": Object {
    "allRowsSelected": true,
    "selectedRows": Array [],
  },
}
`;

exports[`TasksTableReducer reducer should handle SELECT_ROWS 1`] = `
Object {
  "tasksTableQuery": Object {
    "selectedRows": Array [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
    ],
  },
}
`;

exports[`TasksTableReducer reducer should handle TASKS_TABLE_SET_PAGINATION 1`] = `
Object {
  "tasksTableQuery": Object {
    "selectedRows": Array [],
  },
}
`;

exports[`TasksTableReducer reducer should handle TASKS_TABLE_SET_SORT 1`] = `
Object {
  "tasksTableQuery": Object {
    "selectedRows": Array [],
  },
}
`;

exports[`TasksTableReducer reducer should handle TASKS_TABLE_SUCCESS 1`] = `
Object {
  "tasksTableQuery": Object {
    "actionName": undefined,
    "itemCount": 120,
    "pagination": Object {
      "page": 3,
      "perPage": 12,
    },
    "permissions": Object {
      "edit": undefined,
    },
    "selectedRows": Array [],
  },
}
`;

exports[`TasksTableReducer reducer should handle UNSELECT_ALL_ROWS 1`] = `
Object {
  "tasksTableQuery": Object {
    "allRowsSelected": false,
    "selectedRows": Array [],
    "showSelectAll": false,
  },
}
`;

exports[`TasksTableReducer reducer should handle UNSELECT_ROWS 1`] = `
Object {
  "tasksTableQuery": Object {
    "allRowsSelected": false,
    "selectedRows": Array [],
    "showSelectAll": false,
  },
}
`;

exports[`TasksTableReducer reducer should handle UNSELECT_ROWS with all rows selected 1`] = `
Object {
  "tasksTableQuery": Object {
    "allRowsSelected": false,
    "selectedRows": Array [
      3,
      4,
      5,
    ],
    "showSelectAll": false,
  },
}
`;

exports[`TasksTableReducer reducer should handle UPDATE_CLICKED 1`] = `
Object {
  "tasksTableQuery": Object {
    "clicked": "task",
    "selectedRows": Array [],
  },
}
`;

exports[`TasksTableReducer reducer should return the initial state 1`] = `
Object {
  "tasksTableQuery": Object {
    "selectedRows": Array [],
  },
}
`;
